saleor/graphql/discount/mutations/sale_update.py

```diff
@@ -30,16 +33,76 @@ class Meta:
     @classmethod
     @traced_atomic_transaction()
     def perform_mutation(cls, _root, info, **data):
-        node_id = data.get("id")
-        instance = cls.get_node_or_error(info, node_id, only_type=Sale)
+        instance = cls.get_instance(info, **data)
         previous_catalogue = fetch_catalogue_info(instance)
-        response = super().perform_mutation(_root, info, **data)
-        current_catalogue = fetch_catalogue_info(instance)
+        data = data.get("input")
+        cleaned_input = cls.clean_input(info, instance, data)
+        instance = cls.construct_instance(instance, cleaned_input)
+        cls.clean_instance(info, instance)
+        cls.save(info, instance, cleaned_input)
+        cls._save_m2m(info, instance, cleaned_input)
+        cls.send_sale_notifications(info, instance, cleaned_input, previous_catalogue)
+        return cls.success_response(instance)
+
+    @classmethod
+    def clean_input(cls, info, instance, data, input_cls=None):
+        clean_input = super().clean_input(info, instance, data)
+        cls.update_notification_flag_if_needed(instance, clean_input, "start")
```

Comment:
[Question] Wouldn't be better to have one single method for this?  You always call it twice for `start` and `end`, so maybe would be worth to have single method here. Especially that you relay on `field` field. Where `field` is a part of the name of DB field